### ISSUE TITLE
False Positive: MD063 on Numbered Header

### ISSUE THREAD

**Issue by athyfr** [rvben/rumdl#552]
### rumdl version

0.1.58

### Installation method

Other

### Bug description

Given a header with text starting with a list index (i.e. `1.`) followed by a word in the `lowercase-words` list, the word should (intuitively) be capitalized.

### Steps to reproduce

1. Create a Markdown document containing the following text:
    ```md
    ## 1. To Be a Thing
    ```
2. Run `rumdl check <filename>` on the file.
3. See `[MD063]` flag the file.

### Expected behavior

The text:

```md
## 1. To Be a Thing
```

Is flagged by Rumdl, giving the following message:

```
[MD063] Heading should use title case: '1. To Be a Thing' -> '1. to Be a Thing'
```

### Actual behavior

The text:

```md
## 1. To Be a Thing
```

Should not be flagged by `[MD063]` in Rumdl.

### Configuration

```toml
[global]
disable = ["MD045"]
extend-enable = ["MD060", "MD063", "MD072", "MD073"]

# ...
```

### Operating system

Linux

### Additional context

I think the ideal way to solve this would be to require capitalizing words after periods (`.`). This would give another exception to the `lowercase-words` list (in addition to capitalizing them at the start of the header.)

**Comment by rvben** [comment]
Fixed in v0.1.64. Words after a period-terminated token like `1.` are now treated as sentence-initial, so `## 1. To Be a Thing` no longer triggers MD063. Thanks for the clear reproduction case, @athyfr!

**Comment by rvben** [comment]
Fixed in v0.1.65. MD063 now treats any word following a period-terminated token (e.g. `1.`, `2.`) as the start of a new clause, so `## 1. To Be a Thing` is correctly left unchanged.
